File: backend/src/main/java/es/udc/paproject/backend/model/services/InscriptionService.java
package es.udc.paproject.backend.model.services;

import es.udc.paproject.backend.model.entities.Inscription;
import es.udc.paproject.backend.model.exceptions.*;

import java.time.LocalDateTime;

public interface  InscriptionService {

    // [FUNC-3]
    Inscription inscribeUser (Long userId, Long trialId, String cardNumber)
            throws MaxInscriptionsReachedException, DuplicateInscriptionException, DateExpiredException, InstanceNotFoundException;

    // [FUNC-5] Visualización del histórico de inscripciones.
    Block<Inscription> findInscriptionsByUser(Long userId, int page, int size);

    void rateTrial (Long inscriptionId, Long userId, int score) throws
            InstanceNotFoundException, AlreadyScoredException, DateExpiredException, DifferentUserException;

    // [FUNC-4] Entrega de dorsal
    int giveNumber(Long inscriptionId, Long trialId, String cardNumber)
            throws InstanceNotFoundException, DateExpiredException,
            NumberAlreadyDeliveredException, InvalidCreditCardException, IncorrectTrialException;
}

File: frontend/src/backend/inscriptionService.js
import {config, appFetch} from './appFetch';

export const inscribeUser = (trialId, cardNumber, onSuccess, onErrors) =>
    appFetch(`/inscriptions/${trialId}/inscribe`, config('POST', cardNumber), onSuccess, onErrors);

export const giveNumber = (trialId, inscriptionId, cardNumber, onSuccess, onErrors) =>
    appFetch(`/inscriptions/${trialId}/giveNumber`,
        config('POST', {inscriptionId, cardNumber}), onSuccess, onErrors);

export const findInscriptionByUser = ({page}, onSuccess) =>
    appFetch(`/inscriptions/inscriptionHistory?page=${page}` ,
        config('GET'), onSuccess);

export const rateTrial = (inscriptionId, score, onSuccess, onErrors) =>
    appFetch(`/inscriptions/${inscriptionId}/rateTrial`,
        config('POST', score), onSuccess, onErrors);